### ISSUE TITLE
Crash when adding without entered URL in Stream Mode

### ISSUE THREAD

**Issue by NathanBnm** [torikulhabib/mindi#14]
If you add some blank link into Stream mode, downloading is starting but there is nothing to download and the app just doesn't respond anymore. You may add a dialog to avoid that.

Here is a screenshot:

![screenshot 2019-01-19 21 34 20](https://user-images.githubusercontent.com/45366162/51432077-09363200-1c32-11e9-9c93-d0da6680b412.png)

**Comment by ryonakano** [comment]
I think if the entry in the popover is blank, the add button should be insensitive like this:

![peek 2019-01-26 12-52](https://user-images.githubusercontent.com/26003928/51782319-b399e200-2169-11e9-8a4f-faa7a9c94cd4.gif)

**Comment by NathanBnm** [comment]
@ryonakano fixing this will not be enough. We need to check if the entered URL is valid. Because if you enter anything else than a valid URL, Mindi is just freezing.

**Comment by ryonakano** [comment]
@NathanBnm Yes, I also mind that, but if we enter and add invalid string, Mindi shows an error like this and does not freeze:

![peek 2019-01-26 22-49](https://user-images.githubusercontent.com/26003928/51788062-f46f1680-21bc-11e9-8ce3-b0d1043a2772.gif)

So I assumed we don't have to check whether the entered URL is valid.

**Comment by NathanBnm** [comment]
@ryonakano by my side when I enter a invalid string Mindi is just freezing on "Downloading..."

![image](https://user-images.githubusercontent.com/45366162/51788137-f1ebcd00-217a-11e9-8d23-f182b0bd2a13.png)

With no possibility to close the app. The only way is to kill process with Terminal.

**Comment by ryonakano** [comment]
Hmm… :thinking: 

Are you installed Mindi from AppCener or by building source code? I installed by building source code.

**Comment by ryonakano** [comment]
Also, could you open your Terminal, run `com.github.torikulhabib.mindi`, and tell me outputs?

**Comment by NathanBnm** [comment]
> Hmm… thinking
> 
> Are you installed Mindi from AppCener or by building source code? I installed by building source code.

I have installed it from AppCenter. Idk how to build from source for now.

**Comment by NathanBnm** [comment]
I launched it from Terminal. Here is the error that I obtain:

`** (com.github.torikulhabib.mindi:17876): WARNING **: 16:01:49.228: Converter.vala:205: Le changement de répertoire « /home/nathan/.cache/com.github.torikulhabib.mindi » a échoué (No such file or directory)`

**Comment by ryonakano** [comment]
Oh, that should be the reason why the app freezes on your side. I think if you create the directory `/home/nathan/.cache/com.github.torikulhabib.mindi`, the app should show an error if an invalid URL is entered.

Or download a valid video from YouTube, then that directory is automatically created.

**Comment by ryonakano** [comment]
**TL; DR: The app freezes when users enter URLs other than YouTube if users have never download videos from there**

There are a condition which checks the existence of the cache directory and if that directory doesn't exist (`if (!cache_dir.query_exists ()`),  it is created automatically (`cache_dir.make_directory_with_parents ();`):

https://github.com/torikulhabib/mindi/blob/a28c5e6c8f0f983b802f6de18984ea332c6c94e5/src/Objects/Converter.vala#L162-L179

However, this process is only executed if the condition "stream" is true (`if (stream)`).

I checked when that condition is true and when is false. Then I found that if the given URL does not include strings like "youtu" (`if (url.contains ("youtu"))`), the condition "stream" turns false:

https://github.com/torikulhabib/mindi/blob/a28c5e6c8f0f983b802f6de18984ea332c6c94e5/src/Window.vala#L431-L450

This is why the app freeze when you enter an invalid URL.

**Comment by NathanBnm** [comment]
> Oh, that should be the reason why the app freezes on your side. I think if you create the directory `/home/nathan/.cache/com.github.torikulhabib.mindi`, the app should show an error if an invalid URL is entered.
> 
> Or download a valid video from YouTube, then that directory is automatically created.

@ryonakano  Indeed, that is the source of the problem! But in this case **it is necessary to create this folder at the first launch** of the application to avoid any problem.

NB: Thanks for explanation it's really great :+1:

**Comment by ryonakano** [comment]
@NathanBnm Thanks, I've created a new PR which fix this issue!
